homework/jaellysbales/Horoscope/app/src/main/java/com/jaellysbales/horoscope/drawer/FragmentDrawer.java

```diff
@@ -0,0 +1,170 @@
+package com.jaellysbales.horoscope.drawer;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.support.v4.widget.DrawerLayout;
+import android.support.v7.app.ActionBarDrawerToggle;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.Toolbar;
+import android.view.GestureDetector;
+import android.view.LayoutInflater;
+import android.view.MotionEvent;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.jaellysbales.horoscope.R;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class FragmentDrawer extends Fragment {
+
+    private static String TAG = FragmentDrawer.class.getSimpleName();
+
+    private RecyclerView recyclerView;
+    private ActionBarDrawerToggle mDrawerToggle;
+    private DrawerLayout mDrawerLayout;
+    private NavigationDrawerAdapter adapter;
+    private View containerView;
+    private static String[] titles = null;
+    private FragmentDrawerListener drawerListener;
+
+    public FragmentDrawer() {
+
+    }
+
+    public void setDrawerListener(FragmentDrawerListener listener) {
+        this.drawerListener = listener;
+    }
+
+    public static List<NavDrawerItem> getData() {
+        List<NavDrawerItem> data = new ArrayList<>();
+
+
+        // preparing navigation drawer items
```

Comment:
Good use of looping through your array to assign content to your nav drawer rather than adding one by one.